src/main/java/com/limechain/trie/structure/TrieStructure.java

```diff
@@ -495,4 +496,52 @@ public <U> boolean structurallyEquals(TrieStructure<U> other) {
             }
         }
     }
+
+    public boolean removeNode(Nibbles key) {
```

Comment:
[Suggestion] This could become a method on the `NodeHandle` class, thus guaranteeing that it could only be called on existing nodes. Where I see a benefit in that (avoiding the uncertainty which necessitates the `boolean` return type), I might see why one could find it convoluted. Still, that's what I would've done, at least for consistency's sake (the whole idea of the entire `...Handle`s being namely that - mutating operations on nodes, so for me it semantically passes as a method there) :D